gapic/generator/generator.py

```diff
@@ -106,25 +107,59 @@ def get_response(
             )
 
         sample_output = self._generate_samples_and_manifest(
-            api_schema,
-            self._env.get_template(sample_templates[0]),
-        ) if sample_templates else {}
-
+                api_schema, self._env.get_template(sample_templates[0]),
+                opts=opts,
+            )
         output_files.update(sample_output)
 
+
         # Return the CodeGeneratorResponse output.
         res = CodeGeneratorResponse(
             file=[i for i in output_files.values()])  # type: ignore
         res.supported_features |= CodeGeneratorResponse.Feature.FEATURE_PROTO3_OPTIONAL  # type: ignore
         return res
 
+    
+    def _generate_sample_specs(self, api_schema: api.API, *, opts):
+        """Given an API, generate rudimentary sample specs."""
+
+        sample_specs = []
+
+        gapic_metadata = api_schema.gapic_metadata(opts)
+
+        # Loop through the method list
+        for service_name, service in gapic_metadata.services.items():
+            raise Exception(f"duplicate entries for {service.clients}")
+            for client_name, client in service.clients.items():
+                
+                for rpc_name, method_list in client.rpcs.items():
+                    # generated_${api}_${apiVersion}_${serviceName}_${rpcName}_{sync|async}
```

Comment:
This is going to change now so that snippets can be matched to products by the snippet tracker. @busunkim96 I've cc'ed you on a comment on the desgin doc.

Reply:
Fixed! One thing I noticed is that the endpoint is declared at the service level.  I guess in practice the endpoint is the same across all services in the same API? https://github.com/googleapis/googleapis/blob/ae5fb2884a296832c39867e8e8c81bbc72a32ce8/google/cloud/pubsublite/v1/subscriber.proto#L35